bin/common.bash

```diff
@@ -228,3 +228,29 @@ check_openstack_credentials() {
         exit 1
     fi
 }
+
+# Compares the expected and actual git state of the kubespray submodule.
+# If they don't match the user will be asked if they want to continue anyway.
+kubespray_version_check(){
+   pushd "${here}/../" || exit
+
+   result=$(git diff kubespray/)
+
+   popd || exit
+
+   if [[ $result ]]; then
+
+        subproject1=$(echo "${result}" | grep -m1 commit | grep -o '[^ ]*$')
+        subproject2=$(echo "${result}" | tail -n 1 | grep -o '[^ ]*$')
+
+        log_info "Kubespray commit mismatch."
+        log_info "Expected" "${subproject1}", "got" "${subproject2}".
+        log_info_no_newline "Do you want to abort? (y/n): "
+
+        read -r reply
+
+        if [[ "${reply}" == "y" ]]; then
+            exit 1
+        fi
```

Comment:
Also, why did you remove the explanation?

> "The status of the kubespray git submodule differs from the expected status, either it is on another commit or there are file changes. This can cause unexpected versions to be installed or cause other errors. We recommend that you stop and check what has changed."

Reply:
Fixed! :+1: